src/main/java/net/glowstone/chunk/GlowChunk.java

```diff
@@ -755,7 +755,30 @@ public ChunkDataMessage toMessage(boolean skylight, boolean entireChunk) {
         /**
          * The coordinates.
          */
-        private final int x, z;
+        private final int x, z, hashCode;
+
+        private Key(int x, int z) {
+            this.x = x;
+            this.z = z;
+            this.hashCode = x * 31 + z;
+        }
+
+        @Override
+        public int hashCode() {
+            return hashCode;
+        }
     }
 
+    public static final class ChunkKeyStore {
+        private static final ConcurrentHashMap<Integer, Key> keys = new ConcurrentHashMap<>();
+
+        public static Key get(int x, int z) {
+            int id = x * 31 + z;
```

Comment:
Can we maybe remove the duplication of this hashcode generation code?

Reply:
Suee